commercetools-convenience/src/test/java/io/sphere/sdk/meta/CategoryDocumentationIntegrationTest.java

```diff
@@ -73,29 +74,110 @@ public static void beforeClass() throws IOException {
+            .queryAll(client(), CategoryQuery.of(), (categories -> categories), 500);
+        final List<List<Category>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+
+        assertThat(categoryPages).hasSize(1);
+
+        final List<Category> categories = categoryPages.stream()
+                                                       .flatMap(List::stream)
+                                                       .collect(toList());
+
         assertThat(categories)
-                .hasSize(15)
-                .matches(cats -> cats.parallelStream().anyMatch(cat -> cat.getSlug().get(ENGLISH).equals("boots-women")));
+            .hasSize(15)
+            .matches(cats -> cats.parallelStream().anyMatch(cat -> cat.getSlug().get(ENGLISH).equals("boots-women")));
     }
+
+    @Test
+    public void fetchAllExternalIdsWithUniformPageSizes() throws Exception {
+        final Function<List<Category>, List<String>> externalIdCallBack = categories ->
+            categories.stream()
+                      .map(Category::getExternalId)
+                      .collect(toList());
+
+        final CompletionStage<List<List<String>>> categoryPagesStage = QueryExecutionUtils
+            .queryAll(client(), CategoryQuery.of(), externalIdCallBack, 3);
+        final List<List<String>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+
+        assertThat(categoryPages).hasSize(5);
+
+        final List<String> externalIds = categoryPages.stream()
+                                                      .flatMap(List::stream)
+                                                      .collect(toList());
+
+        assertThat(externalIds).hasSize(15);
+        IntStream.range(0, externalIds.size()).forEach(index -> assertThat(externalIds).contains(index + ""));
+    }
+
+    @Test
+    public void fetchAllExternalIdsWithNonUniformPageSizes() throws Exception {
+        final Function<List<Category>, List<String>> externalIdCallBack = categories ->
+            categories.stream()
+                      .map(Category::getExternalId)
+                      .collect(toList());
+
+        final CompletionStage<List<List<String>>> categoryPagesStage = QueryExecutionUtils
+            .queryAll(client(), CategoryQuery.of(), externalIdCallBack, 4);
+        final List<List<String>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+
+        assertThat(categoryPages).hasSize(4);
+
+        final List<String> externalIds = categoryPages.stream()
+                                                      .flatMap(List::stream)
+                                                      .collect(toList());
+
+        assertThat(externalIds).hasSize(15);
+        IntStream.range(0, externalIds.size()).forEach(index -> assertThat(externalIds).contains(index + ""));
```

Comment:
@andrii-kovalenko-ct That's a good idea and I'm the right person to ask 😉 And since it's only used in our tests the update won't cause any problems for our users. I've created a more general issue for updating our test dependencies and you can find it here: #1518